Cython/Compiler/ExprNodes.py

```diff
@@ -8374,16 +8374,22 @@ def generate_special_parallel_unpacking_code(self, code, rhs, use_loop):
         if len(sequence_types) == 2:
             code.putln("if (likely(Py%s_CheckExact(sequence))) {" % sequence_types[0])
         for i, item in enumerate(self.unpacked_items):
-            code.putln("%s = Py%s_GET_ITEM(sequence, %d); " % (
-                item.result(), sequence_types[0], i))
+            if sequence_types[0] == "List":
+                code.putln(f"{item.result()} = __Pyx_PyList_GetItemRef(sequence, {i});")
```

Comment:
We need to do error handling here, in case `__Pyx_PyList_GetItemRef()` returns NULL.